map.js

```diff
@@ -0,0 +1,10 @@
+function map(ob, fun) {
```

Comment:
:ok: but incoming  array should be immutable
create a new array for transformed values and return it